index.html

```diff
@@ -4292,6 +4306,454 @@ <h3>Proof Formats</h3>
+              <li>
+Other extracting functions are possible, but need not be listed in
+the <a data-cite="?VC-SPECS#media-type-extensions">Media Type Extensions</a>
+section of the Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Determine the specific cryptographic verification algorithm
+(<var>verifyProof</var>) required to verify the securing mechanism:
+            <ol class="algorithm">
+              <li>
+If the <var>inputMediaType</var> is `application/vc+ld+json` or
+`application/vp+ld+json`, the Verifiable Credential Data Integrity
+Specification [[?VC-DATA-INTEGRITY]] defines the process for identifying
+a specific cryptography suite type used for cryptographic verification by
+utilizing the `proof`.`type` property. The list of all registered
+proof types can be found in the <a data-cite="?VC-SPECS#proof">Proof Types</a>
+section of the Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+              </li>
+              <li>
+If the <var>inputMediaType</var> is associated with the JOSE or COSE
+specifications, the [[?VC-JOSE-COSE]] specification defines a process for
+identification of a specific cryptography suite and cryptographic verification
+process.
+              </li>
+              <li>
+Other proof verification functions are possible, but need not be listed in
+the <a data-cite="?VC-SPECS#proof">Proof Types</a> section of the
+Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Set <var>securingMechanismVerified</var>.<var>status</var> to the
+result of passing <var>inputBytes</var>, along with any relevant function-specific
+options, to the <var>verifyProof</var> function. For example, for
+Verifiable Credential Data Integrity [[?VC-DATA-INTEGRITY]] proof
+verification functions, the <var>expectedProofPurpose</var> option is set to
+`assertionMethod`, and if the <a>conforming document</a> is expected to be a
+<a>verifiable presentation</a>, the <var>domain</var> and <var>challenge</var>
+are also provided as options.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`false`, add a <a href="#CRYPTOGRAPHIC_SECURITY_ERROR">
+  CRYPTOGRAPHIC_SECURITY_ERROR</a> to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`true`:
+            <ol class="algorithm">
+              <li>
+Set <var>securingMechanismVerified</var>.<var>document</var> by passing
+<var>inputBytes</var> to the <var>extractDocument</var> function.
```

Comment:
If any parts of the input document aren't checked by the proof, this will let them through to consumers. For vc-data-integrity, that means the `proof` field gets through, especially any of its attributes that didn't contribute to actually checking the proof. Ideally, I think https://www.w3.org/TR/vc-data-integrity/#verify-proof should return either the _unsecuredDocument_ or the _transformedData_ for use in the _document_ field, since those are the things it actually verifies.

Reply:
Yes, good point. Fixed in 040950e3326974e0bf65f947c08d9f771c4852e1 (by aligning w/ what happens in vc-data-integrity).

Reply:
Yes, I agree that the current fix isn't as good as we would like... ideally, the securing mechanism defines an algorithm provides only the secured information. That said, I think that's what effectively happens with the change in 040950e, the risk being what you outlined (if mechanisms "evolve" to start letting unsecured data through, that is clearly bad and a failure... and the algorithm will need to be updated to ensure that doesn't happen).

As you can [see here](https://github.com/w3c/vc-data-model/pull/1338#pullrequestreview-1750963355), getting the WG to define proper interfaces is going to be more challenging than we'd like.

Perhaps we can track your concern here: https://github.com/w3c/vc-data-model/issues/1362

... and at the risk of opening a new can of worms, how do you see unsecured data being let through as securing mechanisms evolve? I ask because I don't think there is any variation of Data Integrity that exists today that can pass a signature check AND let unsecured data through. Feel free to put the response in #1362, as we do want to (eventually) do the right thing and improve the first cut at this algorithm.

Reply:
See comment here: https://github.com/w3c/vc-data-model/pull/1338#discussion_r1407889636

`proof` and really the entire conforming document byte stream are integrity checked when JOSE / COSE is used.

You can even write a JSON Schema that forbids certain cryptosuites from being present in the proof fields (there can be multiple proofs).

Reply:
Part of why we're in the pickle we're in now is that we let the "proof" property stay in the VCDM - a legacy decision we inherited from VCDM 1.1.  "proof" is specific to data integrity and should only be specified there.  Having it in VCDM continues to cause confusion.

"proof" isn't part of a Credential.  It's part of one flavor of Verifiable Credentials.  To get this right, we need to move "proof" to DI and remove it from the VCDM.

Reply:
> "proof" is specific to data integrity and should only be specified there.

No, it is not. 

I get that there are people that continue to be confused by this in the WG, and we should fix that, but "proof" was, and still is, an extension point on the VCDM. It just so happens that the Data Integrity specifications are aligned with that extension point, but it doesn't mean that the Data Integrity specifications are the ONLY specifications that are allowed to use that extension point. 

The VCDM requires that the `proof` extension point includes a `type` value, but that's (more or less it). DI takes the concept and runs with it, but that doesn't mean that some other specification couldn't define a completely different way of securing the core data model that utilizes the `proof` extension point in the same way as Data Integrity.

> To get this right, we need to move "proof" to DI and remove it from the VCDM.

No, that is not correct.

We should make it more clear that `proof` is an extension point on the VCDM.

Reply:
> We should make it more clear that `proof` is an extension point on the VCDM.

I have hopes that this might be achieved being clear and consistent about the existence of the `enveloped` and `watermarked` (modulo bikeshedding) subtypes of VCs/VPs. There are probably some points in the document(s) where we now have a single section (of whatever depth) that should be broken into two — one for each subtype.